### PR TITLE
DDF-4086 Update organization and address typos in hardening guide

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
Several sections of the hardening guide were moved to make the process of installing a hardened system more clear. Several typos were found during this process and will be addressed as part of this ticket.

#### Who is reviewing it? 
@austinsteffes @garrettfreibott @ahoffer @ricklarsen @shaundmorris 

#### How should this be tested?
Inspect changes for suggestions and improvements.

#### What are the relevant tickets?
[DDF-4086](https://codice.atlassian.net/browse/DDF-4086)

#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc [hunk 1]
```diff
@@ -50,6 +50,13 @@ To enable guest authentication for a context, use the <<{managing-prefix}configu
 If a legacy client requires the use of the secured SOAP endpoints, the <<{developing-prefix}configuring_guest_interceptor,guest interceptor>> should be configured.
 Otherwise, the guest interceptor and `public` endpoints should be uninstalled for a hardened system.
 
+To uninstall:
```
Comment (ahoffer) [inline distribution/docs/src/main/resources/content/_configuring/configuring-guest-access.adoc]: As a first-time reader, I think line 53 should be even more explicit:

> To uninstall the guest interceptor and public endpoints:

File: distribution/docs/src/main/resources/content/_securing/hardening-intro.adoc [hunk 2]
```diff
@@ -16,7 +16,11 @@ Where available, necessary migitations to harden an installation of ${branding}
 Refer to the <<{managing-prefix}hardening_checklist,Hardening Checklist>> for a compilation of these mitigations.
 ****
 
+[NOTE]
+====
 These security precautions are best performed as configuration is taking place, so hardening steps are integrated into configuration steps.
```
Comment (ahoffer) [inline distribution/docs/src/main/resources/content/_securing/hardening-intro.adoc]: What follows is a list of what should be configured for a hardened system. It is not intended to be a step-by-step procedure. To harden a new system, perform configuration as <<${managing-prefix}configuring.adoc,documented>>.  The configuration procedure includes instructions to hardening ${branding}.

Comment (mcalcote) [reply distribution/docs/src/main/resources/content/_securing/hardening-intro.adoc]: After looking at this together, @ahoffer and I agreed that this info may be more useful as an appendix.

**Comment by mcalcote** [comment]
build now

**Comment by ricklarsen** [comment]
👏 👏

**Comment by mcalcote** [comment]
Looks like the build started having connection refused and timeout issues half way through the itests... build now

**Comment by mcalcote** [comment]
The failing anchor works locally, but I updated it after consulting with @ricklarsen. build now

**Comment by mcalcote** [comment]
build now

**Comment by ricklarsen** [comment]
tested changes manually...

**Pull request merged**